# Data Cleaning and Transformation
*****************************
Understanding of Removing rows, columns & duplicates, Renaming column, Dropping and Filling-in missing values, Creating dummy variables and exporting pandas DataFrame for Data Cleaning and Transformation.

### 1. Removing Columns or Rows

Load the titanic.csv dataset to a dataframe and than print it.

In [1]:
import pandas as pd
df = pd.read_csv('titanic2.csv')
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

Remove the following columns from the dataframe: "SibSp", "Parch", "Ticket". Print the updated dataframe.

In [2]:
df.drop(['SibSp','Parch','Ticket'],axis=1,inplace=True)
df


PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age     Fare  \
0                              Braund, Mr. Owen Harris    male  22.0   7.2500   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0  71.2833   
2                               Heikkinen, Miss. Laina  female  26.0   7.9250   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0  53.1000   
4                             Allen, Mr. William Henry    male  35.0   8.0500   
..                                                 ...     ...   ...      ...   
886                              Montvila, Rev. Juozas    male  27.0  13.0000   
887                       Graham, Miss. Margaret Edith  female  19.0  30.0000   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN  23.4500   
889                              Behr, Mr. Karl Howell    male  26.0  30.0000   
890                                Dooley, Mr. Patrick    male  32.0   7.7500   

    Cabin Embarked  
0     NaN        S  
1     C85        C  
2     NaN        S  
3    C123        S  
4     NaN        S  
..    ...      ...  
886   NaN        S  
887   B42        S  
888   NaN        S  
889  C148        C  
890   NaN        Q  

[891 rows x 9 columns]

Remove the first 20 rows from the dataframe.Print the updated dataframe.

In [3]:
df.drop(range(0,20),inplace=True)
df

PassengerId  Survived  Pclass                                      Name  \
20            21         0       2                      Fynney, Mr. Joseph J   
21            22         1       2                     Beesley, Mr. Lawrence   
22            23         1       3               McGowan, Miss. Anna "Annie"   
23            24         1       1              Sloper, Mr. William Thompson   
24            25         0       3             Palsson, Miss. Torborg Danira   
..           ...       ...     ...                                       ...   
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age     Fare Cabin Embarked  
20     male  35.0  26.0000   NaN        S  
21     male  34.0  13.0000   D56        S  
22   female  15.0   8.0292   NaN        Q  
23     male  28.0  35.5000    A6        S  
24   female   8.0  21.0750   NaN        S  
..      ...   ...      ...   ...      ...  
886    male  27.0  13.0000   NaN        S  
887  female  19.0  30.0000   B42        S  
888  female   NaN  23.4500   NaN        S  
889    male  26.0  30.0000  C148        C  
890    male  32.0   7.7500   NaN        Q  

[871 rows x 9 columns]

### 2. Removing Duplicate Rows 

Reload the titanic.csv dataset to a dataframe and print the LAST five rows.

In [4]:
df = pd.read_csv('titanic2.csv')
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

Add five duplicated rows to the dataframe that are duplicated from the first row. And than add six duplicated rows to the dataframe that are duplicated from the second row (tip - use for-loop and the loc method with 891 offset). 

In [5]:
for i in range(5):
    df = df.append(df.iloc[0],ignore_index=True)

for i in range(6):
    df = df.append(df.iloc[1],ignore_index=True)
    
df

C:\Users\158372\AppData\Local\Temp\ipykernel_20968\3366691629.py:2: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df.iloc[0],ignore_index=True)
C:\Users\158372\AppData\Local\Temp\ipykernel_20968\3366691629.py:5: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df.iloc[1],ignore_index=True)


PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
897            2         1       1   
898            2         1       1   
899            2         1       1   
900            2         1       1   
901            2         1       1   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
897  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
898  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
899  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
900  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
901  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
897      0          PC 17599  71.2833   C85        C  
898      0          PC 17599  71.2833   C85        C  
899      0          PC 17599  71.2833   C85        C  
900      0          PC 17599  71.2833   C85        C  
901      0          PC 17599  71.2833   C85        C  

[902 rows x 12 columns]

Print the last 15 rows.

In [6]:
df.tail(15)

PassengerId  Survived  Pclass  \
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   
891            1         0       3   
892            1         0       3   
893            1         0       3   
894            1         0       3   
895            1         0       3   
896            2         1       1   
897            2         1       1   
898            2         1       1   
899            2         1       1   
900            2         1       1   
901            2         1       1   

                                                  Name     Sex   Age  SibSp  \
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   
891                            Braund, Mr. Owen Harris    male  22.0      1   
892                            Braund, Mr. Owen Harris    male  22.0      1   
893                            Braund, Mr. Owen Harris    male  22.0      1   
894                            Braund, Mr. Owen Harris    male  22.0      1   
895                            Braund, Mr. Owen Harris    male  22.0      1   
896  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
897  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
898  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
899  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
900  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
901  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

     Parch      Ticket     Fare Cabin Embarked  
887      0      112053  30.0000   B42        S  
888      2  W./C. 6607  23.4500   NaN        S  
889      0      111369  30.0000  C148        C  
890      0      370376   7.7500   NaN        Q  
891      0   A/5 21171   7.2500   NaN        S  
892      0   A/5 21171   7.2500   NaN        S  
893      0   A/5 21171   7.2500   NaN        S  
894      0   A/5 21171   7.2500   NaN        S  
895      0   A/5 21171   7.2500   NaN        S  
896      0    PC 17599  71.2833   C85        C  
897      0    PC 17599  71.2833   C85        C  
898      0    PC 17599  71.2833   C85        C  
899      0    PC 17599  71.2833   C85        C  
900      0    PC 17599  71.2833   C85        C  
901      0    PC 17599  71.2833   C85        C

Print the amount of duplicated rows in the dataframe (it should be 11).

In [7]:
df.duplicated().sum()

11

Print the duplicated rows using the loc method combined with the duplicated method.

In [8]:
df[df.duplicated()]

PassengerId  Survived  Pclass  \
891            1         0       3   
892            1         0       3   
893            1         0       3   
894            1         0       3   
895            1         0       3   
896            2         1       1   
897            2         1       1   
898            2         1       1   
899            2         1       1   
900            2         1       1   
901            2         1       1   

                                                  Name     Sex   Age  SibSp  \
891                            Braund, Mr. Owen Harris    male  22.0      1   
892                            Braund, Mr. Owen Harris    male  22.0      1   
893                            Braund, Mr. Owen Harris    male  22.0      1   
894                            Braund, Mr. Owen Harris    male  22.0      1   
895                            Braund, Mr. Owen Harris    male  22.0      1   
896  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
897  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
898  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
899  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
900  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
901  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

     Parch     Ticket     Fare Cabin Embarked  
891      0  A/5 21171   7.2500   NaN        S  
892      0  A/5 21171   7.2500   NaN        S  
893      0  A/5 21171   7.2500   NaN        S  
894      0  A/5 21171   7.2500   NaN        S  
895      0  A/5 21171   7.2500   NaN        S  
896      0   PC 17599  71.2833   C85        C  
897      0   PC 17599  71.2833   C85        C  
898      0   PC 17599  71.2833   C85        C  
899      0   PC 17599  71.2833   C85        C  
900      0   PC 17599  71.2833   C85        C  
901      0   PC 17599  71.2833   C85        C

Print the shape of the dataframe (should be 902,12).

In [9]:
df.shape

(902, 12)

Drop all duplicated lines while keeping the first occurnace of duplicated row and save the changes in the dataframe. Print the shape attribute (should be back to 891,12).

In [10]:
df.drop_duplicates(keep='first')


PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

### 3. Renaming Column Labels 

Reload the titanic.csv dataset to a dataframe.

In [11]:
df = pd.read_csv('titanic2.csv')

Print the columns labels attribute.

In [12]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

Rename the following column labels: 'Name' into 'Full_Name', 'Fare':'Ticket_Price'.

In [13]:
df.rename(columns={'Name':'Full_Name','Fare':'Ticket_Price'},inplace=True)

Print the updated columns labels attribute.

In [14]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Full_Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Ticket_Price', 'Cabin', 'Embarked'],
      dtype='object')

Remove all columns labels to the following new labels: ['C1', 'C2','C3','C4','C5','C6','C7','C8','C9','C10','C11','C12']

In [15]:
df.columns = ['C1', 'C2','C3','C4','C5','C6','C7','C8','C9','C10','C11','C12']

Print the updated columns labels attribute.

In [16]:
df.columns

Index(['C1', 'C2', 'C3', 'C4', 'C5', 'C6', 'C7', 'C8', 'C9', 'C10', 'C11',
       'C12'],
      dtype='object')

### 4. Dropping Missing Values

Reload the titanic.csv dataset to a dataframe.

In [17]:
df = pd.read_csv('titanic2.csv')

Check the number of missing values per each column.

In [18]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

The "Cabin" columns has many missing data rows so let's remove thise column from the dataset. Print again the number of missing values per each column. 

In [19]:
df.drop('Cabin',axis=1,inplace=True)
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         2
dtype: int64

Drop all rows with missing values.

In [20]:
df = df.dropna()
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

### 5. Filling-in Missing Values 

Reload the titanic.csv dataset to a dataframe.

In [21]:
df = pd.read_csv('titanic2.csv')

Fill all missing values in the "Age" column with the average age. 

In [22]:
df['Age'].fillna(df['Age'].mean(),inplace=True)

Print the amount of missing values in each column.

In [23]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

How many rows we have for each value in the 'Embarked' column including the missing values? (should be 2).

In [24]:
df['Embarked'].isnull().sum()

2

Fill the missing values in the 'Embarked' column and also 'Cabin" column with the text value "missing".

In [25]:
df.fillna("missing",inplace=True)

Print the dataframe. 

In [26]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex        Age  \
0                              Braund, Mr. Owen Harris    male  22.000000   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.000000   
2                               Heikkinen, Miss. Laina  female  26.000000   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.000000   
4                             Allen, Mr. William Henry    male  35.000000   
..                                                 ...     ...        ...   
886                              Montvila, Rev. Juozas    male  27.000000   
887                       Graham, Miss. Margaret Edith  female  19.000000   
888           Johnston, Miss. Catherine Helen "Carrie"  female  29.699118   
889                              Behr, Mr. Karl Howell    male  26.000000   
890                                Dooley, Mr. Patrick    male  32.000000   

     SibSp  Parch            Ticket     Fare    Cabin Embarked  
0        1      0         A/5 21171   7.2500  missing        S  
1        1      0          PC 17599  71.2833      C85        C  
2        0      0  STON/O2. 3101282   7.9250  missing        S  
3        1      0            113803  53.1000     C123        S  
4        0      0            373450   8.0500  missing        S  
..     ...    ...               ...      ...      ...      ...  
886      0      0            211536  13.0000  missing        S  
887      0      0            112053  30.0000      B42        S  
888      1      2        W./C. 6607  23.4500  missing        S  
889      0      0            111369  30.0000     C148        C  
890      0      0            370376   7.7500  missing        Q  

[891 rows x 12 columns]

### 6. Creating Dummy Variables

Reload the titanic.csv dataset to a dataframe.

In [27]:
df = pd.read_csv('titanic2.csv')

Remove the following columns: 'Pclass', 'SibSp', 'Parch', 'Ticket', 'Cabin' from the dataframe. Print the updated dataframe. 

In [28]:
df.drop(['Pclass','SibSp','Parch','Ticket','Cabin'],axis=1,inplace =True)
df

PassengerId  Survived                                               Name  \
0              1         0                            Braund, Mr. Owen Harris   
1              2         1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2              3         1                             Heikkinen, Miss. Laina   
3              4         1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4              5         0                           Allen, Mr. William Henry   
..           ...       ...                                                ...   
886          887         0                              Montvila, Rev. Juozas   
887          888         1                       Graham, Miss. Margaret Edith   
888          889         0           Johnston, Miss. Catherine Helen "Carrie"   
889          890         1                              Behr, Mr. Karl Howell   
890          891         0                                Dooley, Mr. Patrick   

        Sex   Age     Fare Embarked  
0      male  22.0   7.2500        S  
1    female  38.0  71.2833        C  
2    female  26.0   7.9250        S  
3    female  35.0  53.1000        S  
4      male  35.0   8.0500        S  
..      ...   ...      ...      ...  
886    male  27.0  13.0000        S  
887  female  19.0  30.0000        S  
888  female   NaN  23.4500        S  
889    male  26.0  30.0000        C  
890    male  32.0   7.7500        Q  

[891 rows x 7 columns]

Drop rows with missing values in the 'Embarked' column.

In [29]:
df.dropna(subset=['Embarked'],inplace=True)

Print the unique values we have in the 'Embarked' column (should be 'S', 'C', 'Q').

In [30]:
df['Embarked'].unique()

array(['S', 'C', 'Q'], dtype=object)

Create dummy variables for the 'Embarked' column using the get_dummies function (it is not a method of dataframe). The 'Embarked' column as 3 possible values for this categorical feature, so we are expecting to get 2 dummy variables (tip - use the drop_first argument). 

In [31]:
df1 = pd.get_dummies(df['Embarked'],drop_first=True)
df1

Q  S
0    0  1
1    0  0
2    0  1
3    0  1
4    0  1
..  .. ..
886  0  1
887  0  1
888  0  1
889  0  0
890  1  0

[889 rows x 2 columns]

Print the updated dataframe.

In [32]:
df = pd.concat([df,df1],axis=1)
df

PassengerId  Survived                                               Name  \
0              1         0                            Braund, Mr. Owen Harris   
1              2         1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2              3         1                             Heikkinen, Miss. Laina   
3              4         1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4              5         0                           Allen, Mr. William Henry   
..           ...       ...                                                ...   
886          887         0                              Montvila, Rev. Juozas   
887          888         1                       Graham, Miss. Margaret Edith   
888          889         0           Johnston, Miss. Catherine Helen "Carrie"   
889          890         1                              Behr, Mr. Karl Howell   
890          891         0                                Dooley, Mr. Patrick   

        Sex   Age     Fare Embarked  Q  S  
0      male  22.0   7.2500        S  0  1  
1    female  38.0  71.2833        C  0  0  
2    female  26.0   7.9250        S  0  1  
3    female  35.0  53.1000        S  0  1  
4      male  35.0   8.0500        S  0  1  
..      ...   ...      ...      ... .. ..  
886    male  27.0  13.0000        S  0  1  
887  female  19.0  30.0000        S  0  1  
888  female   NaN  23.4500        S  0  1  
889    male  26.0  30.0000        C  0  0  
890    male  32.0   7.7500        Q  1  0  

[889 rows x 9 columns]

### 7. Exporting Pandas DataFrames

Save the df dataframe into a file called "output.csv" inside your working directory, without using the index row numbers in the output file.

In [33]:
df.to_csv('output.csv')

View the output.csv file and check that everything is OK!

In [34]:
df2 = pd.read_csv('output.csv')
df2

Unnamed: 0  PassengerId  Survived  \
0             0            1         0   
1             1            2         1   
2             2            3         1   
3             3            4         1   
4             4            5         0   
..          ...          ...       ...   
884         886          887         0   
885         887          888         1   
886         888          889         0   
887         889          890         1   
888         890          891         0   

                                                  Name     Sex   Age     Fare  \
0                              Braund, Mr. Owen Harris    male  22.0   7.2500   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0  71.2833   
2                               Heikkinen, Miss. Laina  female  26.0   7.9250   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0  53.1000   
4                             Allen, Mr. William Henry    male  35.0   8.0500   
..                                                 ...     ...   ...      ...   
884                              Montvila, Rev. Juozas    male  27.0  13.0000   
885                       Graham, Miss. Margaret Edith  female  19.0  30.0000   
886           Johnston, Miss. Catherine Helen "Carrie"  female   NaN  23.4500   
887                              Behr, Mr. Karl Howell    male  26.0  30.0000   
888                                Dooley, Mr. Patrick    male  32.0   7.7500   

    Embarked  Q  S  
0          S  0  1  
1          C  0  0  
2          S  0  1  
3          S  0  1  
4          S  0  1  
..       ... .. ..  
884        S  0  1  
885        S  0  1  
886        S  0  1  
887        C  0  0  
888        Q  1  0  

[889 rows x 10 columns]

### End of document
************************